src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/HKDF.cs

```diff
@@ -0,0 +1,255 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System.Security.Cryptography
+{
+    /// <summary>
+    /// RFC5869  HMAC-based Extract-and-Expand Key Derivation (HKDF)
+    /// </summary>
+    /// <remarks>
+    /// In situations where the input key material is already a uniformly random bitstring, the HKDF standard allows the Extract
+    /// phase to be skipped, and the master key to be used directly as the pseudorandom key.
+    /// See <a href="https://tools.ietf.org/html/rfc5869">RFC5869</a> for more information.
+    /// </remarks>
+    public static class HKDF
+    {
+        /// <summary>
+        /// Performs the HKDF-Extract function.
+        /// See section 2.2 of <a href="https://tools.ietf.org/html/rfc5869#section-2.2">RFC5869</a>
+        /// </summary>
+        /// <param name="hashAlgorithmName">Hash algorithm used for HMAC operations</param>
+        /// <param name="ikm">Input keying material</param>
+        /// <param name="salt">Optional salt value (a non-secret random value). If not provided it defaults to a byte array of <see cref="HashLength"/> zeros.</param>
+        /// <returns>Pseudo random key (prk)</returns>
+        public static byte[] Extract(HashAlgorithmName hashAlgorithmName, byte[] ikm, byte[] salt = null)
+        {
+            if (ikm == null)
+                throw new ArgumentNullException(nameof(ikm));
+
+            int hashLength = HashLength(hashAlgorithmName);
+            ReadOnlySpan<byte> saltSpan = salt ?? ReadOnlySpan<byte>.Empty;
```

Comment:
the null array is assignable to a (ReadOnly)Span, you shouldn't need to coalesce it.

```suggestion
            ReadOnlySpan<byte> saltSpan = salt;
```